### ISSUE TITLE
In multiple database mode, two tables having the same name is an issue on JOINs

### ISSUE THREAD

**Issue by timsehn** [dolthub/dolt#2279]
```
mysql> select
             SHAQ.players.player_id as player_id,
             SHAQ.teams.team_id as team_id,
             left(stats.season_id,4) as season_id,
             0 as season_type_id
     from
             nba_players.season_totals_regular_season as stats
     join
             SHAQ.players
     on
             SHAQ.players.nba_player_id = stats.player_id
     join
     nba_players.teams
     on
     stats.team_id =nba_players.teams.id
     join
     SHAQ.teams
     on
     SHAQ.teams.full_name = nba_players.teams.full_name
     limit 20;
Error 1105: Not unique table/alias: teams
```

**Comment by timsehn** [comment]
This is a good alias bug to think about @fulghum.

**Comment by timsehn** [comment]
Still broken after all our alias changes :-(
```
MySQL [(none)]> select dolt_version();
+----------------+
| dolt_version() |
+----------------+
| 1.16.3         |
+----------------+
1 row in set (0.006 sec)

MySQL [(none)]> select              SHAQ.players.player_id as player_id,              SHAQ.teams.team_id as team_id,              left(stats.season_id,4) as season_id,              0 as season_type_id      from              nba_players.season_totals_regular_season as stats      join              SHAQ.players      on              SHAQ.players.nba_player_id = stats.player_id      join      nba_players.teams      on      stats.team_id =nba_players.teams.id      join      SHAQ.teams      on      SHAQ.teams.full_name = nba_players.teams.full_name      limit 20;
ERROR 1105 (HY000): Not unique table/alias: teams
```

**Comment by timsehn** [comment]
This is fixed by #6866
